src/lightkurve/lightcurve.py

```diff
@@ -3140,7 +3195,113 @@ def plot_river(self, **kwargs):
             period=self.period, epoch_time=self.epoch_time, **kwargs
         )
         return ax
+    
 
+    def bin(
+        self,
+        time_bin_size=None,
+        time_bin_start=None,
+        aggregate_func=None,
+        bins=None,
```

Comment:
- Shouldn't the parameters here be the same as the `bin()` of regular (non-folded) lightcurve?
  - shouldn't `n_bins` (and probably `binsize`) be supported as well?
  - if `time_bin_size` is supported even for a folded lightcurve with normalized phase, shouldn't `time_bin_end` be included as well?

Reply:
Actually, looking into time_bin_end more closely, it looks like it doesn't really get passed in the the regular bin() function. I do want to remove it as an option (I will add that in the change file).

Reply:
Hmm, it does look `time_bin_end` parameter never gets passed in regular bin(), an issue dated from the introduction of the feature in #1042 (v2.1.0). 

I'm confused about what you intend to do:
1. On the one hand, for regular lightcurve, codes are added in this PR to properly pass along `time_bin_end` to the underlying `aggregate_downsample()`.
2. On the other hand, for FoldedLightCurve, `time_bin_end` is added as a parameter but is not passed along.

Reply:
I made the last commit before finishing my edits. I intend to remove time_bin_end as an optional keyword for FoldedLightCurves:
    def bin(
        self,
        time_bin_size=None,
        time_bin_start=None,
        aggregate_func=None,
        bins=None,
        n_bins=None,
    ):